plugin/hooks.go

```diff
@@ -149,4 +153,16 @@ type Hooks interface {
 	// Note that this method will be called for files uploaded by plugins, including the plugin that uploaded the post.
 	// FileInfo.Size will be automatically set properly if you modify the file.
 	FileWillBeUploaded(c *Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string)
+
+	// PushNotificationWillBeSent is invoked before the push notification has been sent to the push proxy server.
+	PushNotificationWillBeSent(c *Context, notification *model.PushNotification) *model.PushNotification
+
+	// PushNotificationHasBeenSent is invoked after the push notification has been sent to the push proxy server.
+	PushNotificationHasBeenSent(c *Context, notification *model.PushNotification)
+
+	// PushNotificationHasFailed is invoked after the push proxy server replies with an error.
```

Comment:
Wording change for consistency

```suggestion
	// PushNotificationHasFailed is invoked after the push proxy server has replied with an error.
```

Reply:
Done